plugins/cairo-lang-macro/src/lib.rs

```diff
@@ -18,17 +18,22 @@
 pub use cairo_lang_macro_attributes::*;
 #[doc(hidden)]
 pub use linkme;
+use std::cell::RefCell;
 
 use cairo_lang_macro_stable::ffi::StableSlice;
 use cairo_lang_macro_stable::{
     StableExpansionsList, StablePostProcessContext, StableProcMacroResult,
 };
 use std::ffi::{c_char, CStr, CString};
+use std::ops::Deref;
 
 mod types;
 
 pub use types::*;
 
+// A thread-local allocation context for allocating tokens on proc macro side.
+thread_local!(static CONTEXT: RefCell<AllocationContext> =  RefCell::default() );
```

Comment:
Could you explain a little bit why do we have to make it `thread_local`?

Reply:
`RefCell` is not `Sync`, this makes access to it easier, if it is not `thread_local` simple `RefCell<T>` must be replaced with `Arc<Mutex<T>>`

Reply:
Exactly this ^

We want this to be available globally, so `Token::new` can access it implicitly.